runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaDoFnRunners.java

```diff
@@ -388,6 +407,7 @@ private void emitMetrics() {
       final long finishBundleTime = System.nanoTime();
       final long averageProcessTime = (finishBundleTime - startBundleTime) / count;
 
+      String metricName = "ExecutableStage-" + stepName + "-process-ns";
```

Comment:
can be extracted to an instance variable since this doesn't seem to change after construction?

Reply:
I don't like the use of instance variables. We already have a large pool of it, and would like to keep only primitive data in there.

On the same note, I want to keep data as local as possible. This `metricName` seems only useful for the emitMetrics method at the moment (it's specific to Executable stage time) , and may conflict with other metric names.

I think putting the computation here improves the code cleanness, and out weights the cost to compute it every time.